src/components/stack-chart/Canvas.js

```diff
@@ -425,44 +465,64 @@ class StackChartCanvas extends React.PureComponent<Props> {
       return;
     }
     const { depth, stackTimingIndex } = hoveredItem;
-    const { stackTimingByDepth, updatePreviewSelection } = this.props;
+    const { combinedTimingRows, updatePreviewSelection } = this.props;
     updatePreviewSelection({
       hasSelection: true,
       isModifying: false,
-      selectionStart: stackTimingByDepth[depth].start[stackTimingIndex],
-      selectionEnd: stackTimingByDepth[depth].end[stackTimingIndex],
+      selectionStart: combinedTimingRows[depth].start[stackTimingIndex],
+      selectionEnd: combinedTimingRows[depth].end[stackTimingIndex],
     });
   };
 
-  _getCallNodeIndexFromHoveredItem(
+  _getCallNodeIndexOrMarkerIndexFromHoveredItem(
     hoveredItem: HoveredStackTiming | null
-  ): IndexIntoCallNodeTable | null {
+  ): {| index: number, type: 'marker' | 'call-node' |} | null {
     if (hoveredItem === null) {
       return null;
     }
 
     const { depth, stackTimingIndex } = hoveredItem;
-    const { stackTimingByDepth } = this.props;
-    const callNodeIndex = stackTimingByDepth[depth].callNode[stackTimingIndex];
-    return callNodeIndex;
+    const { combinedTimingRows } = this.props;
+
+    if (combinedTimingRows[depth].callNode) {
+      const callNodeIndex =
+        combinedTimingRows[depth].callNode[stackTimingIndex];
+      return { index: callNodeIndex, type: 'call-node' };
```

Comment:
I was a bit concerned with returning an object here, but the components appear to be memoizing things correctly and not over-rendering.